core/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -357,6 +469,142 @@ public void register(ActionHandler<?, ?> handler) {
         return unmodifiableList(filters);
     }
 
+    private Set<Class<? extends RestHandler>> setupRestHandlers() {
+        ExtensionPoint.ClassSet<RestHandler> handlers = new ExtensionPoint.ClassSet<>("rest_handler", RestHandler.class);
```

Comment:
As long as we make sure plugins don't try to overwrite any of our actions, I think we will be fine. They pass back a set, so if they decide to insert the same class twice, it's their own silliness.